openpype/hosts/houdini/api/lib.py

```diff
@@ -649,3 +649,43 @@ def get_color_management_preferences():
         "display": hou.Color.ocio_defaultDisplay(),
         "view": hou.Color.ocio_defaultView()
     }
+
+
+def get_obj_node_output(obj_node):
+    """Find output node.
+
+    get the output node with the minimum 'outputidx'
+     or the node with display flag.
+    """
+
+    outputs = obj_node.subnetOutputs()
+    if not outputs:
+        return
+
+    elif len(outputs) == 1:
+        return outputs[0]
+
+    else:
+        return min(outputs,
+                   key=lambda node: node.evalParm('outputidx'))
+
+
+def get_output_children(output_node, include_sops=True):
+    """Recursively return a list of all output nodes
+    contained in this node including this node.
+
+    It works in a similar manner to output_node.allNodes().
+    """
+    out_list = [output_node]
+
+    if output_node.childTypeCategory() == hou.objNodeTypeCategory():
+        for child in output_node.children():
+            out_list += get_output_children(child, include_sops=include_sops)
+
+    elif include_sops and \
+            output_node.childTypeCategory() == hou.sopNodeTypeCategory():
+        out = get_obj_node_output(output_node)
+        if out:
+            out_list += [out]
+
+    return out_list
```

Comment:
Sorry, what is this for? I feel like the naming on this is weird, `get_output_children` but it also always return itself. That is not a child?

Also, if it is similar to `allNodes()` why not use all nodes?

```python
result = {output_node}  # include self
nodes = output_node.allNodes()

obj_category = hou.objNodeTypeCategory()
sop_capegory = hou.sopNodeTypeCategory()

for node in nodes:
    if node.childTypeCategory() == sop_category:
        out = get_obj_node_output(output_node)
        result.add(out)

return list(result)
```
Only difference is that this would recurse into other types than `objNodeTypeCategory`.
Any way, could you explain what your use case is? I noticed you're using it to validate for 'animation' but do you really need to recurse further than just the main output - how does that influence the extraction?

Reply:
`startnode` parameter on FBX ROP accepts obj node or an explicit sop node
![image](https://github.com/ynput/OpenPype/assets/20871534/0fae9a32-b59a-4ab3-9742-9be7bea7df97)

fbx output depends on the start node and  its output 
1. `SopNode`, no dependencies 
2. `ObjNode` and its output 
3. obj sub network, all obj nodes inside it and their outputs 

So, the above function handle these cases as well as handling nested obj subnetworks.
it only returns the output of obj nodes unlike `allNodes()` returns all nodes.

Reply:
> obj sub network, all obj nodes inside it and their outputs

Interesting - so in that case it'd export something else than what the user would be seeing when 'displaying' that obj sub network?

Reply:
I'm using Maya for reference.. 

---

> so in that case it'd export something else than what the user would be seeing when 'displaying' that obj sub network?

it will work as usual, i.e. it will export what the user would be seeing when 'displaying' that obj sub network

Let show you my example: 
I have this subnetwork hierarchy: 
```
*GROUP (Subnetwork ObjNode)
  ├── **Box (ObjNode)
  │        ├── Some nodes we don't care about
  │        └──  *Output node or the node the with the display flag (SopNode)
  ├── **Sphere (ObjNode)
  │        └──  *Output node (SopNode)
  ├── **Gride (ObjNode)
  │        └──  *Output node (SopNode)
  └── **GROUP_2 (Subnetwork ObjNode)
            ├── **Platonic Plate (ObjNode)  
            │        └──  *Output node (SopNode)
            └── **Torus (ObjNode)
                      └──  *Output node (SopNode)
```
So, this function `get_output_children` should retrieve all the important nodes (that I marked with `*` and `**` above) 
I believe any node with `**` should be tested for naming collisions 
I believe any node with `*` or `**` should be tested for Time Dependency 

![image](https://github.com/ynput/OpenPype/assets/20871534/5489a2ed-094f-4bd7-9a97-0f4c89bffdf6)

> fbx rop don't include parent node `GROUP` name in the fbx file, it's whether ignored or named `world_root`

Cube was inside a subnetwork with `Create Root for Subnet` on 
Cube was inside a subnetwork with `Create Root for Subnet` off 
![image](https://github.com/ynput/OpenPype/assets/20871534/fac2418b-deb1-424e-bac0-df28d2bad4b6)

---

I tried to mimic the same scenario in Maya, and it just ignores Groups 
while creating:
![image](https://github.com/ynput/OpenPype/assets/20871534/4d9db145-50d0-4f5b-aa34-89f44693ff19)

After publishing:
![image](https://github.com/ynput/OpenPype/assets/20871534/84dfd3b7-2f88-4740-8a4c-bb1227f9e008)

Let me know what do you think.

Reply:
I hided box and sphere `ObjNode`s in houdini and this is what it looks like in maya 

![image](https://github.com/ynput/OpenPype/assets/20871534/9da11d94-7db8-43fc-9082-8855427e6382)

Reply:
So basically display flag has no influence on the export 👍 Great. Seems like the collector is handling the use cases that the FBX exporter exports.